### PR TITLE
Chore/update generator

### PR DESCRIPTION
**Pull request by smanwaring**
## Description

- Removed “TkComponent” from generated module name
- Revised initial package.json version to 0.0.0
- Added a console.log message to clarify how to name your new pattern
- Simple react component generator uses `styles['short-module-name']` convention
- Generated docs/article is now nested within `docs/articles/pattern-library/content-patters/{short-module-name}/README.md`
- Refactored answers to remove superfluous `camelize` method 

## Motivation and Context

We're getting a lot of new contributions from people using the generator. I made a few updates in order to make the generator clearer and even _more_ convenient to use and also revise the areas that we were constantly asking contributors to refactor (no camelizing styles, module names vs package names, etc). 

## Types of changes

- [ ] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue. you didn't modify existing tests)
- [x] New feature (non-breaking change which adds functionality. you added at least one new test)
- [ ] Breaking change (fix or feature that would cause existing functionality to change. you had to modify existing tests.)
- [ ] Does this introduce a new pattern/package?

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/xogroup/union/blob/master/.github/CONTRIBUTING.md) document.
- [x] I have updated the documentation as needed.
- [x] I have added tests to cover my changes.

## Reviewer Checklist

- [ ] Is change documented in a release draft? (Only if there are plans to merge PR)

### CODE REVIEW

File: packages/tools/generator-union/generators/pattern/answers.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = class Answers {
   constructor(answers) {
     this.patternName = answers.patternName;
     this.moduleName = pascalize(answers.patternName);
-    this.camelizedModuleName = camelize(answers.patternName);
+    this.camelizedModuleName = camelize(answers.patternName).replace('TkComponent', '');
```
Comment (GeorgeTaveras1231) [inline packages/tools/generator-union/generators/pattern/answers.js]: Cool touch

File: packages/tools/generator-union/generators/pattern/index.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ module.exports = class Pattern extends Generator {
   }
 
   promting() {
+    this.log('\nUse dash-case prefixed by tk-component. Example: tk-component-select-box)');
```
Comment (smanwaring) [inline packages/tools/generator-union/generators/pattern/index.js]: Noticed unnecessary parenthesis. I'll remove.

**Comment by smanwaring** [comment]
@GeorgeTaveras1231 made some more updates. These are all the changes I wanted to make.

File: packages/tools/generator-union/generators/pattern/templates/index.jsx [hunk 3]
```diff
@@ -2,8 +2,8 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styles from '<%= packageCssPath %>';
 
-export default function <%= moduleName %>({ greeting = "Hello world" }) {
-  return <div className={styles.<%= camelizedModuleName %>}>{greeting}</div>;
+export default function <%= moduleName %>({ greeting = 'Hello world' }) {
```
Comment (hsbacot) [inline packages/tools/generator-union/generators/pattern/templates/index.jsx]: Is there any reason we're not using `defaultProps`?

https://facebook.github.io/react/docs/typechecking-with-proptypes.html#default-prop-values

**Comment by hsbacot** [comment]
What the motive to shift from accessing styles from a dot property to the `styles['short-module-name']` convention? It feels much more verbose and prohibits the destructuring of specific class on import.

File: packages/tools/generator-union/generators/pattern/templates/index.jsx (hunk 3 above)
Comment (smanwaring) [inline packages/tools/generator-union/generators/pattern/templates/index.jsx]: Great question. This reminds me that we haven't established a consistent mechanism by which props are defaulted. React has the built-in `defaultProps` so it makes sense to make that the default method.

**Comment by smanwaring** [comment]
@hsbacot re: dot property, this was addressed by George here: https://github.com/xogroup/union/pull/219#discussion_r129826942

**Comment by hsbacot** [comment]
@smanwaring missed that, thanks for pointing me there.

**Pull request merged**